tools/distpartitioning/initialize.py

```diff
@@ -0,0 +1,474 @@
+    Edge data as read from files, which is in the following format: 
+        <global_src_id> <global_dst_id> <global_type_eid> <edge_type> <attributes>
+    is converted to the following format in this function:
+        <global_src_id> <global_dst_id> <global_type_eid> <edge_type> <recv_proc>
+
+    Parameters:
+    -----------
+    rank : integer
+        rank of the process
+    params : argparser object
+        argument parser data structure to access command line arguments
+    node_part_ids : numpy array 
+        array of part_ids indexed by global_nid
+
+    Returns:
+    --------
+    numpy ndarray
+        integer node_data with additional columns added
+    numpy ndarray
+        floats, node_features as read from the node features file
+    numpy ndarray
+        integer, edge_data with additional columns added
+    numpy ndarray
+        floats, edge_features are read from the edge feature file
+    """
+    augmted_node_data = []
+    node_data = read_nodes_file(params.input_dir+'/'+params.nodes_file)
+    augmted_node_data = augment_node_data(node_data, node_part_ids)
+    print('Rank: ', rank, ', Completed loading nodes data: ', augmted_node_data.shape)
+
+    augmted_edge_data = []
+    edge_data = read_edges_file(params.input_dir+'/'+params.edges_file)
+    removed_edge_data = read_edges_file(params.input_dir+'/'+params.removed_edges)
+    edge_data = np.vstack((edge_data, removed_edge_data))
+    augmted_edge_data = augment_edge_data(edge_data, node_part_ids)
+    print('Rank: ', rank, ', Completed loading edges data: ', augmted_edge_data.shape)
+
+    node_features = {}
+    node_features = read_node_features_file( params.input_dir+'/'+params.node_feats_file )
+    print('Rank: ', rank, ', Completed loading node features reading from file ', len(node_features))
+
+    edge_features = {}
+    #edge_features = read_edge_features_file( params.input_dir+'/'+params.edge_feats_file )
+    #print( 'Rank: ', rank, ', Completed edge features reading from file ', len(edge_features) )
+
+    return augmted_node_data, node_features, augmted_edge_data, edge_features
+
+def proc_exec(rank, world_size, params):
+    """ 
+    `main` function for each rank in the distributed implementation.
+
+    Parameters: 
+    -----------
+    rank : integer
+        rank of the current process
+    world_size : integer
+        total no. of ranks
+    params : argparser object
+        argument parser structure to access values passed from command line
+    """
+
+    #Read METIS partitions
+    node_part_ids = read_partitions_file(params.input_dir+'/'+params.partitions_file)
+    print('Rank: ', rank, ', Completed loading metis partitions: ', len(node_part_ids))
+
+    #read graph schema, get ntype_map(dict for ntype to ntype-id lookups) and ntypes list
+    schema_map = read_json(params.input_dir+'/'+params.schema)
+    ntypes_map, ntypes = get_node_types(schema_map)
+
+    if rank == 0: 
```

Comment:
Can you put some comments here how rank0 and rank1,2,3 work?

Reply:
Added detailed comments about the functions of rank-0 and non-rank-0 processes  as suggested.